packages/server/lib/cloud/artifacts/upload_artifacts.ts

```diff
@@ -0,0 +1,205 @@
+    }
+
+    debug('protocol report %O', report)
+
+    return {
+      ...acc,
+      protocol: {
+        ...report,
+        error,
+        errorStack,
+      },
+    }
+  }
+
+  return {
+    ...acc,
+    [key]: (key === 'screenshots') ? [...acc.screenshots, report] : report,
+  }
+}
+
+type UploadArtifactOptions = {
+  protocolManager: ProtocolManager
+  videoUploadUrl?: string
+  video?: string // filepath to the video artifact
+  screenshots?: {
+    screenshotId: string
+    path: string
+  }[]
+  screenshotUploadUrls?: {
+    screenshotId: string
+    uploadUrl: string
+  }[]
+  captureUploadUrl?: string
+  protocolCaptureMeta: {
+    url?: string
+    disabledMessage?: string
+  }
+  quiet?: boolean
+  runId: string
+  instanceId: string
+  spec: any
+  platform: any
+  projectId: any
+}
+
+const extractArtifactsFromOptions = async ({
+  video, videoUploadUrl, screenshots, screenshotUploadUrls, captureUploadUrl, protocolCaptureMeta, protocolManager,
+}: Pick<UploadArtifactOptions,
+  'video' | 'videoUploadUrl' |
+  'screenshots' | 'screenshotUploadUrls' |
+  'captureUploadUrl' | 'protocolManager' | 'protocolCaptureMeta'
+>): Promise<IArtifact[]> => {
+  const artifacts: IArtifact[] = []
+
+  if (videoUploadUrl && video) {
+    try {
+      artifacts.push(await createVideoArtifact(video, videoUploadUrl))
+    } catch (e) {
+      debug('Error creating video artifact: %O', e)
+    }
+  }
+
+  debug('screenshot metadata: %O', { screenshotUploadUrls, screenshots })
+  debug('found screenshot filenames: %o', screenshots)
+  if (screenshots?.length && screenshotUploadUrls?.length) {
+    const screenshotArtifacts = await createScreenshotArtifactBatch(screenshotUploadUrls, screenshots)
+
+    screenshotArtifacts.forEach((screenshot) => {
+      artifacts.push(screenshot)
+    })
+  }
+
+  try {
+    const protocolFilePath = protocolManager?.getArchivePath()
+
+    const protocolUploadUrl = captureUploadUrl || protocolCaptureMeta.url
+
+    debug('should add protocol artifact? %o, %o, %O', protocolFilePath, protocolUploadUrl, protocolManager)
+    if (protocolManager && protocolFilePath && protocolUploadUrl) {
+      artifacts.push(await createProtocolArtifact(protocolFilePath, protocolUploadUrl, protocolManager))
+    }
+  } catch (e) {
+    debug('Error creating protocol artifact: %O', e)
+  }
+
+  return artifacts
+}
+
+export const uploadArtifacts = async (options: UploadArtifactOptions) => {
+  const { protocolManager, protocolCaptureMeta, quiet, runId, instanceId, spec, platform, projectId } = options
+
+  const priority = {
+    'video': 0,
+    'screenshots': 1,
+    'protocol': 2,
+  }
+
+  const artifacts = (await extractArtifactsFromOptions(options)).sort((a, b) => {
+    return priority[a.reportKey] - priority[b.reportKey]
+  })
+
+  let uploadReport: UpdateInstanceArtifactsPayload
+
+  if (!quiet) {
+    logUploadManifest(artifacts, protocolCaptureMeta, protocolManager?.getFatalError())
```

Comment:
```suggestion
    logUploadManifest(artifacts, protocolCaptureMeta, protocolManager.getFatalError())
```

Reply:
addressed in a6339bac2acb1bf0fa7fa4994aa80909b418e8ba